src/clojure/nrepl/rich_printer.clj

```diff
@@ -0,0 +1,460 @@
+  (when-some [e (*elide* (MimeContent. mk-in))]
+    {:content (tagged-literal 'unrepl/... e)}))
+
+(def ^:dynamic *object-representations*
+  "map of classes to functions returning their representation component (3rd item in #unrepl/object [class id rep])"
+  {clojure.lang.IDeref
+   (fn [x]
+     (let [pending? (and (instance? clojure.lang.IPending x) ; borrowed from https://github.com/brandonbloom/fipp/blob/8df75707e355c1a8eae5511b7d73c1b782f57293/src/fipp/ednize.clj#L37-L51
+                         (not (.isRealized ^clojure.lang.IPending x)))
+           [ex val] (when-not pending?
+                      (try [false @x]
+                           (catch Throwable e
+                             [true e])))
+           failed? (or ex (and (instance? clojure.lang.Agent x)
+                               (agent-error x)))
+           status (cond
+                    failed? :failed
+                    pending? :pending
+                    :else :ready)]
+       {:unrepl.ref/status status :unrepl.ref/val val}))
+
+   clojure.lang.AFn
+   (fn [x]
+     (-> x class .getName main/demunge))
+
+   java.io.File (fn [^java.io.File f]
+                  (into {:path (.getPath f)}
+                        (when (.isFile f)
+                          {:attachment (tagged-literal 'unrepl/mime
+                                                       (into {:content-type "application/octet-stream"
+                                                              :content-length (.length f)}
+                                                             (mime-content #(java.io.FileInputStream. f))))})))
+
+   java.awt.Image (latent-fn [^java.awt.Image img]
+                             (let [w (.getWidth img nil)
+                                   h (.getHeight img nil)]
+                               (into {:width w, :height h}
+                                     {:attachment
+                                      (tagged-literal 'unrepl/mime
+                                                      (into {:content-type "image/png"}
+                                                            (mime-content #(let [bos (java.io.ByteArrayOutputStream.)]
+                                                                             (when (javax.imageio.ImageIO/write
+                                                                                    (doto (java.awt.image.BufferedImage. w h java.awt.image.BufferedImage/TYPE_INT_ARGB)
+                                                                                      (-> .getGraphics (.drawImage img 0 0 nil)))
+                                                                                    "png" bos)
+                                                                               (java.io.ByteArrayInputStream. (.toByteArray bos)))))))})))
+
+   Object (fn [x]
+            (if (-> x class .isArray)
+              (seq x)
+              (str x)))})
+
+; datafy and friends shoud be but to good use there
```

Comment:
Top-level comments are supposed to have 2+ `;`. At least according to common Lisp lore. :-)